nncf/pruning/filter_pruning/algo.py

```diff
@@ -281,62 +287,129 @@ def _find_layerwise_pruning_rate(self, target_flops_pruning_rate):
+                            # Looking for layerwise pruning rate needed for the required flops pruning rate
+                            pruning_rate = self._find_uniform_pruning_rate_for_target_flops(pruning_rate)
+                    self._set_binary_masks_for_pruned_modules_groupwise(pruning_rate)
 
             if self.zero_grad:
                 self.zero_grads_for_pruned_modules()
+
         self._apply_masks()
+
+        if not groupwise_pruning_rates_set:
+            self._pruning_rate = passed_pruning_rate
+        else:
+            self._pruning_rate = self._calculate_global_weight_pruning_rate()
+
         if run_batchnorm_adaptation:
             self.run_batchnorm_adaptation(self.config)
 
-    def _set_binary_masks_for_filters(self, pruning_rate):
-        nncf_logger.debug("Setting new binary masks for pruned modules.")
-
-        with torch.no_grad():
-            for group in self.pruned_module_groups_info.get_all_clusters():
-                filters_num = torch.tensor([get_filters_num(minfo.module) for minfo in group.nodes])
-                assert torch.all(filters_num == filters_num[0])
-                device = group.nodes[0].module.weight.device
-
-                cumulative_filters_importance = torch.zeros(filters_num[0]).to(device)
-                # 1. Calculate cumulative importance for all filters in group
-                for minfo in group.nodes:
-                    filters_importance = self.filter_importance(minfo.module.weight,
-                                                                minfo.module.target_weight_dim_for_compression)
-                    cumulative_filters_importance += filters_importance
-
-                # 2. Calculate threshold
-                num_of_sparse_elems = get_rounded_pruned_element_number(cumulative_filters_importance.size(0),
-                                                                        pruning_rate)
-                threshold = sorted(cumulative_filters_importance)[min(num_of_sparse_elems, filters_num[0] - 1)]
-                mask = calculate_binary_mask(cumulative_filters_importance, threshold)
-
-                # 3. Set binary masks for filter
-                for minfo in group.nodes:
-                    pruning_module = minfo.operand
-                    pruning_module.binary_filter_pruning_mask = mask
+    def _calculate_global_weight_pruning_rate(self) -> float:
+        full_param_count = 0
+        pruned_param_count = 0
+        for minfo in self.pruned_module_groups_info.get_all_nodes():
+            layer_param_count = sum(p.numel() for p in minfo.module.parameters() if p.requires_grad)
+            layer_weight_pruning_rate = self.pruning_rate_for_weight(minfo)
+            full_param_count += layer_param_count
+            pruned_param_count += layer_param_count * layer_weight_pruning_rate
+        return pruned_param_count / full_param_count
+
+    @property
+    def current_groupwise_pruning_rate(self) -> Dict[int, float]:
```

Comment:
I wonder why it is not called `get_*` because this is exactly the getter functionality?